lib/Sema/CSSimplify.cpp

```diff
@@ -8950,10 +8947,12 @@ bool ConstraintSystem::resolveClosure(TypeVariableType *typeVar,
   assignFixedType(typeVar, closureType, closureLocator);
 
   // If there is a result builder to apply, do so now.
+  auto openedClosureResultType =
+      openOpaqueTypeRec(closureType->getResult(), closureLocator);
```

Comment:
It looks like we're opening opaque types before each call to `matchResultBuilder`. Would it be better to pass in only the unopened type and open it before the constraint is generated at the end of `matchResultBuilder`?